### PR TITLE
Update contracts

### PR DESCRIPTION
**Pull request by NIC619**

### CODE REVIEW

**Comment by 108356037** [comment]
I've tested the new deployed `NonStandardAccount`, works as expected! 👍

**Review by 108356037 — DISMISSED** [review]

File: contracts/SignatureAccount.sol [hunk 1]
```diff
@@ -31,10 +34,13 @@ contract SignatureAccount is IAccount {
         //
         // HINT: OpenZeppelin `ECDSA` library has a `tryRecover(hash, signature)` function to recover the signature signer.
 
+        (bool success, ) = payable(entryPoint).call{ value: missingAccountFunds }("");
+        (success);
```
Comment (pilagod) [inline contracts/SignatureAccount.sol]: I think this is better to put before TODO

File: contracts/InitCode.sol [hunk 2]
```diff
@@ -7,58 +7,43 @@ import { UserOperation } from "aa/interfaces/UserOperation.sol";
...
-        return new SampleAccount{ salt: bytes32(salt) }(owner);
+        return new SignatureAccount{ salt: bytes32(salt) }(entryPoint, owner);
     }
 
     function getAccountAddress(uint256 salt, address owner) public view returns (address) {
         return
             Create2.computeAddress(
                 bytes32(salt),
-                keccak256(abi.encodePacked(type(SampleAccount).creationCode, abi.encode(owner)))
+                keccak256(abi.encodePacked(type(SignatureAccount).creationCode, abi.encode(entryPoint), abi.encode(owner)))
```
Comment (pilagod) [inline contracts/InitCode.sol]: nit: Maybe `abi.encode(entryPoint, owner)`?

File: contracts/SignatureAccount.sol [hunk 3]
```diff
@@ -31,10 +34,13 @@ contract SignatureAccount is IAccount {
...
         return SIG_VALIDATION_SUCCEEDED;
     }
 
     function execute(address target, uint256 value, bytes calldata data) external {
+        require(msg.sender == entryPoint, "SignatureAccount: Unauthorized caller");
```
Comment (pilagod) [inline contracts/SignatureAccount.sol:44]: Do we want to add this to every account? The `DepositAccount` seems missing. Or maybe we should extract a base account with this `execute` function for all accounts.

Comment (NIC619) [reply contracts/SignatureAccount.sol:44]: I cover this check in SignatureAccount section so I added it to the example accounts thereafter.

**Review by 108356037 — APPROVED** [review]

**Pull request merged**